src/util/command_line_handling.cpp

```diff
@@ -666,8 +666,10 @@ namespace hpx { namespace util
             }
         }
 
-        if ((vm.count("hpx:connect") || mode_ == hpx::runtime_mode_connect) &&
-            hpx_host == "127.0.0.1")
+        if ((mode_ == hpx::runtime_mode_console || 
+             mode_ == hpx::runtime_mode_connect || 
+             mode_ == hpx::runtime_mode_worker) 
+            && hpx_host == "127.0.0.1")
```

Comment:
It wouldn't do any harm, probably. Incidentally, we should check for `hpx_host == "localhost"` as well, though.